File: src/main/java/edu/kit/iti/formal/stvs/model/table/Commentable.java
package edu.kit.iti.formal.stvs.model.table;

import javafx.beans.property.StringProperty;


/**
 * <p>
 * The interface for anything that has a {@link StringProperty} with the intent of being used as a
 * comment (for example tables ({@link SpecificationTable}), cells ({@link ConstraintCell}) or rows
 * ({@link SpecificationRow})).
 * </p>
 *
 * @author Benjamin Alt
 */
public interface Commentable {

  void setComment(String comment);

  String getComment();

  StringProperty commentProperty();
}

File: stvs/src/main/kotlin/edu/kit/iti/formal/stvs/model/table/Commentable.kt
package edu.kit.iti.formal.stvs.model.table

import javafx.beans.property.StringProperty

/**
 *
 *
 * The interface for anything that has a [StringProperty] with the intent of being used as a
 * comment (for example tables ([SpecificationTable]), cells ([ConstraintCell]) or rows
 * ([SpecificationRow])).
 *
 *
 * @author Benjamin Alt
 */
interface Commentable {
    var comment: String?
        get() = commentProperty.value
        set(value) = commentProperty.set(value)

    val commentProperty: StringProperty
}
